### DISCUSSION TITLE
Table datakey and sort on nested values

### DISCUSSION THREAD

**Discussion by jkanel** [rsuite/rsuite#2129] [Q&A]
How can we implement a `datakey` and sort on a nested value in the row data.  For example, we might want to display the `modify.name` property and make it sortable:

```
{
  id: "asdfasdfasdfasfasdfas",
  name: "ASDASDASDASDASD",
  modify: {
    name: "Susan Smith"
  }
}
```

Thanks.
Jeff K.

**Comment by SevenOutman** [comment]
Hi @jkanel ,

You can provide a render prop to `<Cell>` to customize its output

```jsx
<Cell dataKey="modify.name">
  {row => row.modify.name}
</Cell>
```

But I think it's good if `dataKey` supports nested paths. WDYT @simonguo?

**Reply by jkanel** [reply]
Hi @SevenOutman,

I tried rendering, as you suggested above, however the component does not permit sorting for nested properties.  The sortable icon appears, but it won't sort.

Any other suggestions?  Thanks for considering adopting this functionality into the code base.

**Reply by SevenOutman** [reply]
> The sortable icon appears, but it won't sort.

`<Table>` does not sort the rows directly, but emits the sorting column key via `onSortColumn` callback. You have to manage to sort the rows yourself. Here is a working example https://rsuitejs.com/components/table/#sort

**Reply by simonguo** [reply]
> But I think it's good if dataKey supports nested paths

We added to the plan, this is a simple feature.

**Reply by jkanel** [reply]
We'll follow that approach @SevenOutman.   Looking forward to seeing this added as a datakey feature.

Thanks for your responsiveness!

**Reply by simonguo** [reply]
Nested paths are supported in the latest version.
